commands/ps/type.js

```diff
@@ -38,7 +38,19 @@ Types: ${cli.color.yellow(formation.map((f) => f.type).join(', '))}`)
 
   let displayFormation = co.wrap(function * () {
     let formation = yield heroku.get(`/apps/${app}/formation`)
+    let shielded = yield heroku.get(`/apps/${app}`)
+
+    shielded = shielded.space && shielded.space.shield
     formation = sortBy(formation, 'type')
+
+    formation.forEach((d) => {
+      if (shielded && d.size === 'Private-M') {
+        d.size = 'Shield-M'
```

Comment:
can we generalize this as something like `replace(s.size, 'Private-', 'Shield-')?